### PR TITLE
log snark-work reassignment

### PR DESCRIPTION
**Pull request by deepthiskumar**
Log when snark works are not received within `work-reassignment-wait`

### CODE REVIEW

**Comment by jkrauska** [comment]
@deepthiskumar it is possible to identify the node the job was assigned to?

we want to track which nodes are slow workers, not just that the work didn't come back.

I couldn't tell if this had that...

File: src/lib/work_selector/test.ml [hunk 1]
```diff
@@ -22,6 +22,7 @@ struct
     let p = 50 in
     let snark_pool = T.Snark_pool.create () in
     let fee = Currency.Fee.zero in
+    let logger = Logger.create () in
```
Comment (psteckler) [inline src/lib/work_selector/test.ml]: for test code, should this be `Logger.null ()` ?

File: src/lib/work_selector/test.ml [hunk 2]
```diff
@@ -40,9 +41,12 @@ struct
     Backtrace.elide := false ;
     let snark_pool = T.Snark_pool.create () in
     let fee = Currency.Fee.zero in
+    let logger = Logger.create () in
```
Comment (psteckler) [inline src/lib/work_selector/test.ml]: same

File: src/lib/work_selector/test.ml [hunk 3]
```diff
@@ -91,6 +95,7 @@ struct
     Backtrace.elide := false ;
     let my_fee = Currency.Fee.of_int 2 in
     let p = 50 in
+    let logger = Logger.create () in
```
Comment (psteckler) [inline src/lib/work_selector/test.ml]: same

**Comment by deepthiskumar** [comment]
> @deepthiskumar it is possible to identify the node the job was assigned to?
> 
> we want to track which nodes are slow workers, not just that the work didn't come back.
> 
> I couldn't tell if this had that...

We don't keep track of that information currently.

**Pull request merged**